src/com.target.lorri.varlink

```diff
@@ -12,6 +12,38 @@ type ShellNix (
   path: string
 )
 
+# Monitor the daemon. The method will reply with an update whenever a build begins or ends.
+# Montior will immediately reply with a snapshot of known projects, then a marker event,
+# indicating that the stream of events is now "live."
+method Monitor() -> (event: Event)
+
+type Event (
```

Comment:
Varlink does not provide proper enums or gRPC-like `oneof`. But I think that events can be modelled in a slightly more "type-safe" way. What do you think about this sort of structure:

```
type Event {
    kind: (section_end, started),
    section_end: SectionEnd?, # set iff kind == section_end
    started: Started?, # set iff kind == started
    # ...
}
```

where `SectionEnd`, `Started`, etc. have only non-optional fields?

Reply:
I think this might be interesting to experiment, but I think I need to put it into a followup PR.